domain/model/service/modelservice_test.go

```diff
@@ -10,28 +10,130 @@ import (
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	cons := constraints.Value{
+		Arch:     ptr("amd64"),
+		CpuCores: ptr(uint64(4)),
+		Mem:      ptr(uint64(1024)),
+		RootDisk: ptr(uint64(1024)),
+	}
+	s.mockModelState.EXPECT().GetModelConstraints(gomock.Any()).Return(cons, nil)
+
+	svc := NewModelService(modeltesting.GenModelUUID(c), nil, s.mockModelState)
+	result, err := svc.GetModelConstraints(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, gc.DeepEquals, cons)
+}
+
+func (s *modelServiceSuite) TestGetModelConstraintsFailedModelNotFound(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	s.mockModelState.EXPECT().GetModelConstraints(gomock.Any()).Return(constraints.Value{}, modelerrors.NotFound)
+
+	svc := NewModelService(modeltesting.GenModelUUID(c), nil, s.mockModelState)
+	_, err := svc.GetModelConstraints(context.Background())
+	c.Assert(err, jc.ErrorIs, modelerrors.NotFound)
+}
+
+func (s *modelServiceSuite) TestSetModelConstraints(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	cons := constraints.Value{
+		Arch:     ptr("amd64"),
+		CpuCores: ptr(uint64(4)),
+		Mem:      ptr(uint64(1024)),
+		RootDisk: ptr(uint64(1024)),
+	}
+	s.mockModelState.EXPECT().SetModelConstraints(gomock.Any(), cons).Return(nil)
+
+	svc := NewModelService(modeltesting.GenModelUUID(c), nil, s.mockModelState)
+	err := svc.SetModelConstraints(context.Background(), cons)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *modelServiceSuite) TestSetModelConstraintsFailedSpaceNotFound(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	cons := constraints.Value{
+		Arch:     ptr("amd64"),
+		CpuCores: ptr(uint64(4)),
+		Mem:      ptr(uint64(1024)),
+		RootDisk: ptr(uint64(1024)),
+		Spaces:   ptr([]string{"space1"}),
+	}
+	s.mockModelState.EXPECT().SetModelConstraints(gomock.Any(), cons).Return(networkerrors.SpaceNotFound)
+
+	svc := NewModelService(modeltesting.GenModelUUID(c), nil, s.mockModelState)
+	err := svc.SetModelConstraints(context.Background(), cons)
+	c.Assert(err, jc.ErrorIs, networkerrors.SpaceNotFound)
+}
+
+func (s *modelServiceSuite) TestSetModelConstraintsFailedModelNotFound(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	cons := constraints.Value{
+		Arch:     ptr("amd64"),
+		CpuCores: ptr(uint64(4)),
+		Mem:      ptr(uint64(1024)),
+		RootDisk: ptr(uint64(1024)),
+	}
+	s.mockModelState.EXPECT().SetModelConstraints(gomock.Any(), cons).Return(modelerrors.NotFound)
+
+	svc := NewModelService(modeltesting.GenModelUUID(c), nil, s.mockModelState)
+	err := svc.SetModelConstraints(context.Background(), cons)
+	c.Assert(err, jc.ErrorIs, modelerrors.NotFound)
+}
+
+type legacyModelServiceSuite struct {
+	testing.IsolationSuite
+
 	controllerState *dummyControllerModelState
 	modelState      *dummyModelState
 	controllerUUID  uuid.UUID
 }
 
-var _ = gc.Suite(&modelServiceSuite{})
+var _ = gc.Suite(&legacyModelServiceSuite{})
```

Comment:
Can this change be explained in the patch description?

Reply:
Done & Done